plugins/scale/scale-title-overlay.cpp

```diff
@@ -0,0 +1,309 @@
+#include "scale-title-overlay.hpp"
+
+#include <wayfire/opengl.hpp>
+#include <wayfire/util/log.hpp>
+#include <wayfire/plugins/common/cairo-util.hpp>
+#include <wayfire/plugins/common/simple-texture.hpp>
+
+/**
+ * Class storing an overlay with a view's title, added to scale's transformer.
+ */
+class view_title_overlay_t
+{
+  protected:
+    wf::cairo_text_t overlay;
+    wf::cairo_text_t::params par;
+    /* the transformer we are attached to */
+    wf::scale_transformer_t& tr;
+    /* if text potentially overflows the available space */
+    bool overflow = false;
+
+    /**
+     * Get the transformed bounding box of the view, including the
+     * current transform
+     */
+    wlr_box get_transformed_bounding_box()
+    {
+        wlr_box box = tr.get_transformed_view()->get_wm_geometry();
+        box = tr.trasform_box_without_padding(box);
+        return box;
+    }
+
+    void render_overlay(const wlr_box& box)
+    {
+        par.max_size = {box.width, box.height};
+        auto view = tr.get_transformed_view();
+        auto res  = overlay.render_text(view->get_title(), par);
+        overflow = res.width > overlay.tex.width;
+        view->damage();
+    }
+
+    void render_overlay()
```

Comment:
nitpick: a better name would be `update_overlay_texture` or similar. `render_overlay` sounds like you are actually rendering it on the screen.